docs/user-guide/src/ipam/ipam.md

```diff
@@ -38,15 +36,103 @@ objects.
 
 ## IPAM components
 
-* **IPPool**: a set of IP addresses pools to be used for IP address allocations
-* **IPClaim**: a request for an IP address allocation
-* **IPAddress**: an IP address allocation
+* **IPPool**: A set of IP addresses pools to be used for IP address allocations
+* **IPClaim**: Request for an IP address allocation
+* **IPAddress**: IP address allocation
 
-## Deployment
+### IPPool
 
-You can deploy IPAM using `make deploy` or deploy an example pool using
-`make deploy-examples`. You can also run IPAM controller locally using
-`make run`.
+Example of IPPool:
 
-For more information about this controller and related repositories, see
-[metal3.io](http://metal3.io/).
+```yaml
+apiVersion: ipam.metal3.io/v1alpha1
+kind: IPPool
+metadata:
+  name: pool1
+  namespace: default
+spec:
+  clusterName: cluster1
+  namePrefix: test1-prov
+  pools:
+    - start: 192.168.0.10
+      end: 192.168.0.30
+      prefix: 25
+      gateway: 192.168.0.1
+    - subnet: 192.168.1.1/26
+    - subnet: 192.168.1.128/25
+  prefix: 24
+  gateway: 192.168.1.1
+  preAllocations:
+    claim1: 192.168.0.12
+```
+
+The *spec* field contains the following fields :
+
+* **clusterName**: Name of the cluster to which this pool belongs, it is used to verify whether the resource is paused.
+* **namePrefix**: The prefix used to generate the IPAddress.
+* **pools**: List of IP address pools
+* **prefix**: Default prefix for this IPPool
+* **gateway**: Default gateway for this IPPool
+* **preAllocations**: Default preallocated IP address for this IPPool
+
+The *prefix* and *gateway* can be overridden per pool. Here is the pool definition:
+
+* **start**: IP range start address and it can be omitted if **subnet** is set.
+* **end**: IP range end address and can be omitted.
+* **subnet**: Subnet for the allocation and can be omitted if **start** is set. It is used to verify that the allocated address belongs to this subnet.
+* **prefix**: Override of the default prefix for this pool
+* **gateway**: Override of the default gateway for this pool
+
+### IPClaim
+
+An IPClaim is an object representing a request for an IP address allocation.
+
+Example of IPClaim pool:
+
+```yaml
+apiVersion: ipam.metal3.io/v1alpha1
+kind: IPClaim
+metadata:
+  name: test1-controlplane-template-0-provisioning-pool
+  namespace: default
+spec:
+  pool:
+    name: pool1
+    namespace: default
+```
+
+The *spec* field contains the following :
+
+* **pool**: This is a reference to the IPPool that is requested for
+
+## IPAddress
+
+An IPAddress is an object representing an IP address allocation.
```

Comment:
Maybe add here that the IPAddress would normally be created by IPAM to fill an IPClaim, so the user does not have to create it manually.